cmake/Platform/Windows/Packaging/Tests/test_installer.py

```diff
@@ -0,0 +1,204 @@
+    result = context.run([str(o3de_path),'create-project','--project-path', str(context.project_path)])
+    assert result.returncode == 0, f"o3de.bat failed to create a project with exit code {result.returncode}"
+
+    project_json_path = Path(context.project_path) / 'project.json'
+    assert project_json_path.is_file(), f"No project.json found at {project_json_path}"
+
+
+@pytest.fixture(scope="session")
+def test_compile_project_fixture(test_create_project_fixture, context):
+    """ Project can be configured and compiled. """
+    project_name = Path(context.project_path).name
+    cmake_path = next(context.cmake_runtime_path.glob('**/cmake.exe'))
+    launcher_target = f"{project_name}.GameLauncher"
+
+    # configure
+    result = context.run([str(cmake_path),'-B', str(context.project_build_path), '-S', '.', '-G', 'Visual Studio 16 2019'], cwd=context.project_path)
+    assert result.returncode == 0
+    assert (context.project_build_path / f'{project_name}.sln').is_file()
+
+    # build profile, release is not yet supported in the current installer
+    result = context.run([str(cmake_path),'--build', str(context.project_build_path), '--target', launcher_target, 'Editor', '--config', 'profile','--','-m'], cwd=context.project_path)
+    assert result.returncode == 0
+    assert (context.project_bin_path / f'{launcher_target}.exe').is_file()
+
+
+@pytest.fixture(scope="session")
+def test_run_asset_processor_batch_fixture(test_compile_project_fixture, context):
+    """ AssetProcessorBatch can process assets before running Editor and Launcher tests which use them. """
+    try:
+        result = context.run([str(context.engine_bin_path / 'AssetProcessorBatch.exe'),f'--project-path="{context.project_path}"','/platform=pc'], timeout=30*60)
+        assert result.returncode == 0, f"AssetProcessorBatch.exe failed with exit code {result.returncode}"
+    except TimeoutExpired as e:
+        pass
+
+
+@pytest.fixture(scope="session")
+def test_run_editor_fixture(test_run_asset_processor_batch_fixture, context):
+    """ Editor can be run without crashing. """
+    # write out the attribution shown regset to avoid the modal popup stalling editor load
+    aws_attribution_regset = {
+            "Amazon": {
+                "AWS": {
+                    "Preferences": {
+                        "AWSAttributionConsentShown": True,
+                        "AWSAttributionEnabled": False 
+                    }
+                }
+            }
+        }
+    aws_attribution_path = context.project_path / 'user/Registry/editor_aws_preferences.setreg'
+    with aws_attribution_path.open('w') as f:
+        json.dump(aws_attribution_regset, f) 
+    
+    try:
+        # run Editor.exe for 2 mins 
+        result = context.run([str(context.engine_bin_path / 'Editor.exe'),f'--project-path="{context.project_path}"','--rhi=null','--skipWelcomeScreenDialog=True','+wait_seconds','10','+quit'], timeout=2*60)
```

Comment:
Instead of writing out a file during the test which has the possibility of failure, these settings should be supplied to the subprocess command via the `--regset="/Amazon/AWS/Preferences/AWSAttributionConsentShown=True"` option